plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -132,10 +137,13 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
 
         TupleDomain<ColumnHandle> oldDomain = handle.getConstraint();
         TupleDomain<ColumnHandle> newDomain = oldDomain.intersect(constraint.getSummary());
-
+        List<String> newConstraintExpressions;
         TupleDomain<ColumnHandle> remainingFilter;
+        Optional<ConnectorExpression> remainingExpression;
         if (newDomain.isNone()) {
+            newConstraintExpressions = ImmutableList.of();
             remainingFilter = TupleDomain.all();
+            remainingExpression = Optional.of(Constant.TRUE);
```

Comment:
Why is this set to `TRUE`? Isn't the original expression == remaining expression?

Reply:
newDomain is NONE, so nothing else matters

Reply:
I understand that ANDing TRUE with remaining filter gets correct result. But I think this contradicts the javadoc on ConstraintApplicationResult which says that if original expression == remaining expression we should pass empty optional for this parameter.

Reply:
Optional means that the expression is not consumed (e.g. connector does not support expression pushdown at all).
Here, it can be consumed, easily, because it doesn't matter.